index.html

```diff
@@ -957,6 +959,191 @@ <h4>Error Response</h4>
         </pre>
+        <p>If a <code>readmultipleproperties</code> operation only partially fails (i.e. some <a>Properties</a> can be
+          read but others can not), a <a>Thing</a> MAY include the successfully read values in the error
+          response, in a <code>values</code> member keyed by <a>Property</a> name.
+        </p>
+      </section>
+    </section>
+
+    <section id="writeallproperties">
+      <h3><code>writeallproperties</code></h3>
+      <section id="writeallproperties-request">
+        <h4>Request</h4>
+        <p>To set the value of all <a>Properties</a> of a <a>Thing</a> at once, a <a>Consumer</a> MUST
+          send a
+          <a href="#websocket-messages">message</a> to the <a>Thing</a> which contains the following members:
+        </p>
+        <table class="def">
+          <caption>Members of a <code>writeallproperties</code> request message</caption>
+          <thead>
+            <tr>
+              <th>Member</th>
+              <th>Type</th>
+              <th>Assignment</th>
+              <th>Description</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>messageType</code></td>
+              <td>string</td>
+              <td>"request"</td>
+              <td>A string which denotes that this message is a request sent from a Consumer to a Thing.</td>
+            </tr>
+            <tr>
+              <td><code>operation</code></td>
+              <td>string</td>
+              <td>"writeallproperties"</td>
+              <td>A string which denotes that this message relates to a <code>writeallproperties</code> operation.
+              </td>
+            </tr>
+            <tr>
+              <td><code>values</code></td>
+              <td>object</td>
+              <td>Mandatory</td>
+              <td>An object keyed by <a>Property</a> name, containing the desired value of each <a>Property</a>, with a
+                type and structure conforming to the data schema of each corresponding PropertyAffordance in the
+                <a>Thing Description</a>.
+              </td>
+            </tr>
+          </tbody>
+        </table>
+        <pre class="example" title="writeallproperties request message">
+          {
+            "thingID": "https://mythingserver.com/things/mylamp1",
+            "messageID": "60c33fc4-ed99-446e-a2e3-f3b5be82c34b",
+            "messageType": "request",
+            "operation": "writeallproperties",
+            "values": {
+              "on": true,
+              "level": 75
+            },
+            "correlationID": "551aaa52-ff95-4fb5-a657-2b5f430d47ff"
+          }
+        </pre>
+        <p>When a <a>Thing</a> receives a message from a <a>Consumer</a> with <code>messageType</code> set to
+          <code>request</code> and <code>operation</code> set to <code>writeallproperties</code> it MUST
+          attempt to write the values of all writeable <a>Properties</a> using the values specified in the
```

Comment:
maybe I am thinking too much but given that there is nothing about this in the error messages, here it is: Should the message contain all writeable properties or not? If there is a third writeable property "foo", should the Thing take what it can and provide a "warning" message back or should the request fail?

Reply:
Given the operation is called write _all_ properties I think the request should fail. I have clarified the error response to explicitly say this.